File: examples/java/sdk-reference/observability/basic-usage.java
import software.amazon.lambda.durable.DurableContext;
import software.amazon.lambda.durable.DurableHandler;

public class BasicUsage extends DurableHandler<Object, String> {
    @Override
    public String handleRequest(Object event, DurableContext context) {
        context.getLogger().info("Starting workflow");

        String result = context.step("process", String.class, stepCtx -> "done");

        context.getLogger().info("Workflow complete: {}", result);
        return result;
    }
}

File: examples/csharp/sdk-reference/observability/basic-usage.cs
using Amazon.Lambda.Core;
using Amazon.Lambda.DurableExecution;
using Microsoft.Extensions.Logging;

public class BasicUsageExample
{
    public Task<DurableExecutionInvocationOutput> Handler(
        DurableExecutionInvocationInput input, ILambdaContext context)
        => DurableFunction.WrapAsync<object, string>(Workflow, input, context);

    private async Task<string> Workflow(object input, IDurableContext ctx)
    {
        ctx.Logger.LogInformation("Starting workflow");

        string result = await ctx.StepAsync(
            async (_, _) => "done",
            name: "process");

        ctx.Logger.LogInformation("Workflow complete: {Result}", result);
        return result;
    }
}
